force-app/main/default/classes/DonationHistoryController_TEST.cls

```diff
@@ -0,0 +1,120 @@
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Donation
+* @description Class for test list donation history controller.
+*/
+@IsTest
+public class DonationHistoryController_TEST {
+    @IsTest
+    public static void testGetDonationHistory() {
+        
+        Date myDateTime = Date.newInstance(2020, 2, 2);
+
+        Account mockDefaultAccount = new Account(Name = 'Default Account');
+        insert mockDefaultAccount;
+
+        Contact mockDefaultContact = new Contact(
+            FirstName='Alice', 
+            LastName='Patterson', 
+            AccountId=mockDefaultAccount.Id
+        );
+        Contact secondaryContact = new Contact(
+            FirstName='Josh', 
+            LastName='Patterson',
+            AccountId=mockDefaultAccount.Id
+        );
+        insert mockDefaultContact;
+        insert secondaryContact;
+
+        Opportunity opportunityHardCredit = new Opportunity(
+            Name='HardCredit1', 
+            AccountId=mockDefaultAccount.Id, 
+            Primary_Contact__c=mockDefaultContact.Id, 
+            Amount=400, 
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        Opportunity opportunitySoftCredit = new Opportunity(
+            Name='SoftCredit1', 
+            AccountId=mockDefaultAccount.Id, 
+            Primary_Contact__c=secondaryContact.Id, 
+            Amount=200, 
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        Opportunity opportunityPartialSoftCredit = new Opportunity(
+            Name='PartialSoftCredit1', 
+            AccountId=mockDefaultAccount.Id, 
+            Primary_Contact__c=secondaryContact.Id, 
+            Amount=150, 
+            StageName='Closed Won', 
+            CloseDate=myDateTime
+        );
+        insert opportunityHardCredit;
+        insert opportunitySoftCredit;
+        insert opportunityPartialSoftCredit;
```

Comment:
This should be done in a single DML statement, not 3.

Reply:
done